modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMStructureLocalServiceImpl.java

```diff
@@ -2289,6 +2343,28 @@ private void _validate(
 		}
 	}
 
+	private void _validateExternalReferenceCode(
+		String externalReferenceCode, long structureId, long groupId,
+		long classNameId) {
+
+		if (Validator.isNull(externalReferenceCode)) {
+			return;
+		}
+
+		DDMStructure structure = ddmStructurePersistence.fetchByERC_G_C(
+			externalReferenceCode, groupId, classNameId);
+
+		if ((structure != null) &&
+			(structure.getStructureId() != structureId)) {
+
+			throw new DuplicateDDMStructureExternalReferenceCodeException(
+				StringBundler.concat(
+					"Duplicate structure external reference code \"",
+					externalReferenceCode, "\" for class name ID \"",
+					classNameId, "\" in group \"", groupId, "\""));
```

Comment:
"Duplicate **DDM** structure external reference code \""

Reply:
As we're inside the service layer where there is only one kind of structure, I thought we would consider maintain this pattern without the prefix DDM. Doing the change rn

Reply:
Done